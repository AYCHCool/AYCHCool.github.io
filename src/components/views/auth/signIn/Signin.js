import React, { useState, Fragment, useEffect } from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { signInAccount } from '@Redux/actions/authActions';
import InputField from '@Common/form/InputField';
import LeftDiv from '@Common/auth/leftDiv';
import SocialButtons from '@Common/auth/buttons';
import { getToken } from '@Redux/actions/socialActions';
import { signinText } from '@Common/auth/leftDivText';
import '@Common/auth/auth.scss';

const Signin = (props) => {
  const [values, setValues] = useState({
    email: '',
    password: '',
  });
  const { loading, history, onSubmit } = props;

  useEffect(() => {
    const { location } = props;
    const tokenString = location && location.search;
    if (tokenString) {
      const token = getToken(tokenString);
      const userData = null;
      onSubmit(userData, history, token);
      history.push('/');
    }
  }, [props]);

  const onChange = (event) => {
    event.persist();
    setValues(prevState => ({ ...prevState, [event.target.name]: event.target.value }));
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    const accountVerificationToken = null;
    onSubmit(values, history, accountVerificationToken);
  };
  return (
    <Fragment>
      <div className="container-body">
        <LeftDiv text={signinText} />
        <div className="right-block">
          <div className="img-mobile-field">
            <img src="https://res-console.cloudinary.com/mazus/thumbnails/transform/v1/image/upload/Y19zY2FsZSx3XzI3MQ==/v1567070037/YXRob3JzX2hhdmVuXzFfbDFueG9o/template_primary" alt="Author's Haven Logo" />
          </div>
          <form onSubmit={handleSubmit} className="main-form">
            <div>
              <div className="input-field">
                <InputField
                  type="email"
                  value={values.email}
                  onChange={onChange}
                  name="email"
                  placeholder="Email"
                  className="form-input"
                  required
                />
              </div>
              <div className="input-field">
                <InputField
                  type="password"
                  value={values.password}
                  onChange={onChange}
                  name="password"
                  placeholder="Password"
                  className="form-input"
                  required
                />
              </div>
              <div className="form-links">
                <h5>
                  <Link to="/forgot-password">forgot password?</Link>
                </h5>
              </div>
            </div>
            <div className="input-field">
              <button className="bttn bttn-primary btn-block bttn-large" type="submit" value="Sign In" disabled={loading}>{
                loading ? 'Loading...' : 'Sign In'
              }
                <i className="arrow-forward material-icons">arrow_forward</i>
              </button>
            </div>
          </form>
          <div className="input-field">
            <h2 id="htwo-space">
              <div className="input-row">
                <div className="column1"><hr /></div>
                <div className="column2">OR</div>
                <div className="column1"><hr /></div>
              </div>
            </h2>
          </div>
          <SocialButtons />
          <div className="input-field btn-top-margin navigate-link">
            <h3>New to Authors Haven?
              <Link to="/signup">&nbsp;Sign Up</Link>
            </h3>
          </div>
        </div>
        <div id="clear" />
      </div>
    </Fragment>
  );
};

Signin.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
  loading: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.auth.errors,
  loading: state.auth.loading,
});
const mapDispatchToProps = dispatch => ({
  onSubmit: (userData, history, token) => dispatch(signInAccount(userData, history, token)),
});
export const SigninComponent = Signin;
export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(Signin));
