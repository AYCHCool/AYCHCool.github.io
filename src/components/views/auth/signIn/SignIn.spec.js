import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '@Redux/store/index';
import InputField from '@Common/form/InputField';
import Signin, { SigninComponent } from './Signin';

const props = {
  loading: false,
  history: {
    push: jest.fn(),
  },
  value: 'Some value',
};

describe('SignIn Component', () => {
  it('should render without errors', () => {
    const component = shallow(
      <Router>
        <Signin store={store} {...props} />
      </Router>,
    );
    expect(component).toMatchSnapshot();
  });
  it('should render along with children component', () => {
    const component = mount(
      <Router>
        <Signin store={store} {...props} />
      </Router>,
    );
    expect(component).toMatchSnapshot();
    expect(component.find('button')).toHaveLength(3);
    expect(component.find('input')).toHaveLength(2);
  });

  it('email Input check', () => {
    const onChange = jest.fn();
    const handleSubmit = jest.fn();
    const event = {
      persist: jest.fn(),
      target: { value: 'dave@gmail.com', name: 'email' },
    };
    const wrapper = shallow(<SigninComponent onSubmit={handleSubmit} onChange={onChange} {...props} />);
    wrapper.find('Input[type="email"]').simulate('change', event);
    expect(wrapper.find('input[name="email"]')).toMatchSnapshot();
  });

  it('Password Input check', () => {
    const onChange = jest.fn();
    const handleSubmit = jest.fn();
    const event = {
      persist: jest.fn(),
      target: { value: 'P455word', name: 'password' },
    };
    const wrapper = shallow(<SigninComponent onSubmit={handleSubmit} onChange={onChange} {...props} />);
    wrapper.find('Input[type="password"]').simulate('change', event);
    expect(wrapper.find('input[name="password"]')).toMatchSnapshot();
  });
  it('should call onChange prop with input value', () => {
    const onChange = jest.fn();
    const component = mount(<InputField value="custom email" type="email" name="email" onChange={onChange} />);
    expect(component).toMatchSnapshot();
  });
  it('should call onSubmit prop function when form is submitted', () => {
    const handleSubmit = jest.fn();
    const wrapper = mount(<SigninComponent onSubmit={handleSubmit} {...props} />);
    const form = wrapper.find('form');
    form.simulate('submit');
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
