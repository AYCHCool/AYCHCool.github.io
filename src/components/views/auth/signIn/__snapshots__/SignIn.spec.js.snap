// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Component should render without errors 1`] = `ShallowWrapper {}`;
